### PR TITLE
:bar_chart: Upload ODA data processed by ONE

### PR DESCRIPTION
**Pull request by paarriagadap**

### CODE REVIEW

**Comment by paarriagadap** [comment]
Hi @Marigold! Is there a way to make this server public showing the changes I implemented?

I tried copying this branch (`git branch oda-one-public`): https://github.com/owid/etl/pull/3454, but the staging check fails because of the `ValueError: Variables used in charts will not be deleted automatically` error, which is solved here.

**Comment by Marigold** [comment]
@paarriagadap to made it work, I had to sync all charts from this branch to the new one with
```
etl chart-sync oda-one-data-updated oda-one-public --dry-run
```
(when I was confident, I removed the `--dry-run` flag)

Then I retried buildkite and the error got fixed. Now your charts are available publicly at e.g. https://staging-site-oda-one-public.tail6e23.ts.net/grapher/foreign-aid-received-by-channel.

We have a guide in [ETL docs](https://docs.owid.io/projects/etl/guides/sharing-external/?h=). I'll add a section with `chart-sync`.

File: snapshots/one/2024-10-02/oda_data_extract.py [hunk 1]
```diff
@@ -0,0 +1,294 @@
+"""
```
Comment (paarriagadap) [inline snapshots/one/2024-10-02/oda_data_extract.py:1]: Hi @Marigold! I came back to this script and can't make it run in the ETL. It doesn't find oda_data even if I run `pip install oda_data --upgrade`. In the staging server it finds an incompatibility with `frictionless`:
```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
etl 0.1.0 requires frictionless[pandas]<5.0.0,>=4.40.8, but you have frictionless 5.18.0 which is incompatible.
Successfully installed bblocks-1.4.0 camelot-py-0.11.0 country-converter-1.3 frictionless-5.18.0 hdx-python-country-3.8.5 hdx-python-utilities-3.7.4 humanize-4.11.0 ijson-3.3.0 imf-reader-1.1.0 jsonlines-4.0.0 jsonpath_ng-1.7.0 libhxl-5.2.2 loguru-0.7.2 oda-reader-1.0.0 oda_data-1.5.0 opencv-python-4.10.0.84 ply-3.11 pyarrow-16.1.0 pydeflate-2.1.0 pyjstat-2.4.0 pypdf-3.17.4 pyphonetics-0.5.3 python-io-wrapper-0.3.1 rapidfuzz-3.10.1 ratelimit-2.2.1 requests-file-2.1.0 tableschema-to-template-0.0.13 thefuzz-0.22.1 webdriver-manager-4.0.2 wheel-0.45.1 xlrd3-1.1.0 xlsx2csv-0.8.4 xlsxwriter-3.2.0 xlwt-1.3.0
```

Comment (Marigold) [reply snapshots/one/2024-10-02/oda_data_extract.py:1]: @paarriagadap I've upgraded frictionless, could you rebase / merge and try again, please?

Comment (Marigold) [reply snapshots/one/2024-10-02/oda_data_extract.py:1]: I was able to run it on a staging server with
```
ssh owid@owid@staging-site-oda-one-data-updated
cd etl
uv run pip install oda_data --upgrade
uv run python snapshots/one/2024-10-02/oda_data_extract.py
```
aren't you missing `uv run` when running it locally?

Comment (paarriagadap) [reply snapshots/one/2024-10-02/oda_data_extract.py:1]: Yes, I can run it on staging.

Locally, for me running with and without `uv run` is the same thing for me. Should it not be that case? I automatically activate the environment and I can always do that, also when we had poetry

Comment (Marigold) [reply snapshots/one/2024-10-02/oda_data_extract.py:1]: Yes, it should be the same, I just thought that the error could be related to using different python environment.

Is it enough if you can run it on staging for now?

Comment (paarriagadap) [reply snapshots/one/2024-10-02/oda_data_extract.py:1]: Hi, it's ok, I can run it on staging. Thanks!

**Comment by lucasrodes** [comment]
@paarriagadap what's the status of this PR? I see it might have been left open to share the staging server with external people. If so, maybe you can add "(shared externally)" to the PR title.

(no urgency, just doing some 🧹)

**Comment by paarriagadap** [comment]
@lucasrodes Yes, both this one (and mostly https://github.com/owid/etl/pull/3454) are created to share info publicly. This has been a tricky issue that I haven't had the time to check, but I want to do it these two days and decide if I close them or not.

BTW, I keep track of all the PRs I have opened here https://github.com/owid/etl/pulls/@me, so no worries

**Comment by paarriagadap** [comment]
@lucasrodes Yeah, at least in these cases I always rebase/merge from master + solve conflicts to avoid most problems

**Comment by paarriagadap** [comment]
I am closing this issue, given that we can't extract a good breakdown for sectors, or at least better than the official OECD aggregates we included in chart like [this one](https://ourworldindata.org/grapher/foreign-aid-given-by-sector). More context for this in the main Foreign aid issue: https://github.com/owid/owid-issues/issues/1620#issuecomment-2539504818

**Pull request closed without merging**